### PR TITLE
Return 404 when no annotations found

### PR DESCRIPTION
**Pull request by mchompalova**

### CODE REVIEW

File: handler/handler.go [hunk 1]
```diff
@@ -301,6 +301,9 @@ func (h *Handler) prepareUPPAnnotations(ctx context.Context, contentUUID string,
 
 	ann, err := h.annotationsAPI.GetAllButV2(ctx, contentUUID)
 	if err != nil {
+		if err.Error() == annotations.UPPNotFoundMsg {
+			return nil, http.StatusNotFound, err
+		}
```
Comment (ivan-p-nikolov) [inline handler/handler.go:306]: A better solution would be to propagate all errors from upp and not just the `not found`
```suggestion
		var uppErr *annotations.UPPError
		if errors.As(err, &uppErr) {
			return nil,uppErr.Status(), err
		}
```

Comment (karsov) [reply handler/handler.go:306]: I'm not sure that the general approach is better.
For example, we don't want to return "user error 400" if UPP returns 400, it's not the user's error but an internal error.

Comment (ivan-p-nikolov) [reply handler/handler.go:306]: Fair. Still lets use `errors.As` and then check if it's not found error. Because `err` returned by `annotationsAPI` could be wrapped. And lets compare the `uppErr.Status() == http.StatusNotFound` instead of the message value. This sounds cleaner to me.

File: handler/handler_test.go [hunk 2]
```diff
@@ -1505,6 +1505,31 @@ func TestUnHappyDeleteAnnotationsWhenRetrievingAnnotationsFails(t *testing.T) {
 	assert.Equal(t, http.StatusInternalServerError, resp.StatusCode)
 }
 
+func TestUnHappyDeleteAnnotationsWhenNoAnnotationsFound(t *testing.T) {
+	rw := new(RWMock)
+	annAPI := new(AnnotationsAPIMock)
+	annAPI.On("GetAllButV2", mock.Anything, "83a201c6-60cd-11e7-91a7-502f7ee26895").
+		Return([]annotations.Annotation{}, errors.New(annotations.UPPNotFoundMsg))
```
[the remark this answers is not in the document]
Comment (mchompalova) [reply handler/handler_test.go:1512]: I can't create a new UPPError here because its fields are not publicly visible.

File: handler/handler.go (hunk 1 above)
Comment (mchompalova) [reply handler/handler.go:306]: I also wanted to check Status() but unfortunately errors.As(err, &uppErr) returns false.

File: annotations/augmenter_test.go [hunk 3]
```diff
@@ -158,7 +159,12 @@ func TestAugmentAnnotationsFixtures(t *testing.T) {
 
 			expectedAnnotations := helperGetAnnotationsFromFixture(t, "augmenter-expected-"+test.fixtureBaseName)
 
-			assert.Equal(t, annotations, expectedAnnotations)
+			sort.Slice(annotations, func(i, j int) bool {
+				return annotations[i].Predicate < annotations[j].Predicate
+			})
+			sort.Slice(expectedAnnotations, func(i, j int) bool {
+				return expectedAnnotations[i].Predicate < expectedAnnotations[j].Predicate
+			})
```
Comment (karsov) [inline annotations/augmenter_test.go:167]: The comparison is missing?

File: annotations/testdata/augmenter-input-multiple-predicates-per-concept.json [hunk 4]
```diff
@@ -3,6 +3,10 @@
 		"predicate": "http://www.ft.com/ontology/annotation/about",
 		"id": "http://www.ft.com/thing/b224ad07-c818-3ad6-94af-a4d351dbb619"
 	},
+	{
```
Comment (karsov) [inline annotations/testdata/augmenter-input-multiple-predicates-per-concept.json:6]: It is invalid to have mentions and about for the same concept. Why is this added?

**Comment by karsov** [comment]
Also "This branch is out-of-date with the base branch"

File: annotations/testdata/augmenter-input-multiple-predicates-per-concept.json (hunk 4 above)
Comment (mchompalova) [reply annotations/testdata/augmenter-input-multiple-predicates-per-concept.json:6]: For test purposes, to check with more than 2 elements in the array.
Ok, I can change the predicate.

File: annotations/augmenter_test.go (hunk 3 above)
Comment (mchompalova) [reply annotations/augmenter_test.go:167]: Fixed.
